tests/benchmarks/stitch-lh/src/Language/Stitch/LH/Step.hs

```diff
@@ -73,35 +73,39 @@ substIndex ctx i es e0 = case e0 of
     IntE _ -> e0
     BoolE _ -> e0
 
+{-@ wellTyped :: e:WellTypedExp Nil -> { r:WellTypedExp Nil | e = r } @-}
+wellTyped :: Exp -> Exp
+wellTyped e = e
+
 {-@
 step :: e:WellTypedExp Nil -> Maybe ({ r:WellTypedExp Nil | exprType e = exprType r })
 @-}
 step :: Exp -> Maybe Exp
 step e0 = case e0 of
     Lam{} -> Nothing
     App e1 e2 -> case step e1 of
-      Just e1' -> Just (App e1' e2)
+      Just e1' -> Just (wellTyped (App e1' e2))
```

Comment:
@nikivazou, this is the place I was mentioning where liquid variables interfere with PLE.

PLE gets a left hand side like
`{ VV:_ | numFreeVars VV = 0 && k_var#1234 }`
The `k_var#1234` is useless to PLE, and in addition, it really needs
`{ VV:_ | numFreeVars VV = 0 && VV = App e1' e2 }`
because no binding in the environment relates `VV` to the expression being checked otherwise.